### ISSUE TITLE
Fix `selector-type-*`, `selector-max-type` and `selector-class-pattern` false positives for `<timeline-range-name>` in keyframe selectors

### ISSUE THREAD

**Issue by romainmenke** [stylelint/stylelint#7855]
Scroll driven animations extended the valid syntax for keyframe selectors:
- https://developer.mozilla.org/en-US/docs/Web/CSS/@keyframes
- https://drafts.csswg.org/scroll-animations-1/#named-timeline-range

### What minimal example or steps are needed to reproduce the bug?

```css
@keyframes animate-in-and-out {
	entry 0%  {
		opacity: 0; transform: translateY(100%);
	}

	entry 100%  {
		opacity: 1; transform: translateY(0);
	}

	exit 0% {
		opacity: 1; transform: translateY(0);
	}
	
	exit 100% {
		opacity: 0; transform: translateY(-100%);
	}
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-standard"
  ]
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBrGBPAZgJwIYFsYBnAAlwDsBLfXAFxgFpLyGKoGIBXWk4AHXJ965WtnQkADAFISvAUKEQADrjCVa6eJIDcJURSKYI2fFv3kiAGzowAmgAoAjBOkBKbfNoBfeYOGjxZ2lZfj9FFTUNLUddc0NjUz08C2t6Bwl3Tx9BPxgAD3VJGVCFWmVVdU0SGKSDIxMzZKsbdMy-bIVcgp4g4s9wiqidWot6xLjUu3sGXrahbK8QABoQTEpLGAA5AjhEfIIlDYA6MCIiZfAIcjWAcwQQUNk+EHzhKCJnrQBtAVknkCIGg2lmYtAYkGulBuDEBbFw2Cgz1+JAAugJFisIbcAGIJOj3ABWRCuF1gSnOiEeJGegPQwNBn2pIEmgOeS2RNKBMBBInBV1uMNocIRjOeLNoSPIiy8QA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}
```

### What did you expect to happen?

No problems to be reported

### What actually happened?

> ERROR Unexpected unknown type selector "entry" [(selector-type-no-unknown)](https://stylelint.io/user-guide/rules/selector-type-no-unknown) [2:2-7]
ERROR Unexpected unknown type selector "0%" [(selector-type-no-unknown)](https://stylelint.io/user-guide/rules/selector-type-no-unknown) [2:8-10]
ERROR Unexpected unknown type selector "entry" [(selector-type-no-unknown)](https://stylelint.io/user-guide/rules/selector-type-no-unknown) [6:2-7]
ERROR Unexpected unknown type selector "100%" [(selector-type-no-unknown)](https://stylelint.io/user-guide/rules/selector-type-no-unknown) [6:8-12]
ERROR Unexpected unknown type selector "exit" [(selector-type-no-unknown)](https://stylelint.io/user-guide/rules/selector-type-no-unknown) [10:2-6]
ERROR Unexpected unknown type selector "0%" [(selector-type-no-unknown)](https://stylelint.io/user-guide/rules/selector-type-no-unknown)[10:7-9]
ERROR Unexpected unknown type selector "exit" [(selector-type-no-unknown)](https://stylelint.io/user-guide/rules/selector-type-no-unknown) [14:2-6]
ERROR Unexpected unknown type selector "100%" [(selector-type-no-unknown)](https://stylelint.io/user-guide/rules/selector-type-no-unknown) [14:7-11]

### Do you have a proposal to fix the bug?

Add support for the new pattern of `<timeline-range-name> <percentage>` in `isKeyframeSelector`

**Comment by skyclouds2001** [comment]
see https://github.com/mdn/data/pull/954
